### PR TITLE
Modifying cnos-facts, cnos_command and cnos-config in line with the design followed in Ansible. Adding unit test cases for these modules. Added plugins to support them.

### PR DESCRIPTION
**Pull request by amuraleedhar**
##### SUMMARY
Changing the cnos_facts, cnos_command in tune with the existing functionalities for other modules. Also introducing cnos-config in the same line. 

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/modules/network/cnos/cnos_command.py
lib/ansible/modules/network/cnos/cnos_config.py
lib/ansible/modules/network/cnos/cnos_facts.py
lib/ansible/module_utils/network/cnos/cnos.py
lib/ansible/plugins/action/cnos.py
lib/ansible/plugins/action/cnos_config.py
lib/ansible/plugins/cliconf/cnos.py
lib/ansible/plugins/terminal/cnos.py
test/units/modules/network/cnos/__init__.py
test/units/modules/network/cnos/cnos_module.py
test/units/modules/network/cnos/fixtures/cnos_config_config.cfg
test/units/modules/network/cnos/fixtures/cnos_config_src.cfg
test/units/modules/network/cnos/fixtures/display_interface_brief
test/units/modules/network/cnos/fixtures/display_interface_mac-address
test/units/modules/network/cnos/fixtures/display_interface_status
test/units/modules/network/cnos/fixtures/display_ip_interface_brief_vrf_all
test/units/modules/network/cnos/fixtures/display_ipv6_interface_brief_vrf_all
test/units/modules/network/cnos/fixtures/display_lldp_neighbors
test/units/modules/network/cnos/fixtures/display_process_memory
test/units/modules/network/cnos/fixtures/display_run
test/units/modules/network/cnos/fixtures/display_running-config
test/units/modules/network/cnos/fixtures/display_sys-info
test/units/modules/network/cnos/fixtures/display_version
test/units/modules/network/cnos/test_cnos_command.py
test/units/modules/network/cnos/test_cnos_config.py
test/units/modules/network/cnos/test_cnos_facts.py

##### ANSIBLE VERSION
ansible 2.6.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/sha                                                                                        re/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansibl                                                                                        e-2.6.0-py2.7.egg/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.6 (default, Nov 23 2017, 15:49:48) [GCC 4.8.4]

### CODE REVIEW

File: lib/ansible/modules/network/cnos/cnos_command.py [hunk 1]
```diff
@@ -1,166 +1,228 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
-from __future__ import (absolute_import, division, print_function)
-__metaclass__ = type
-#
-# Copyright (C) 2017 Lenovo, Inc.
 #
-# This file is part of Ansible
+# (C) 2017 Red Hat Inc.
```
Comment (gundalow) [inline lib/ansible/modules/network/cnos/cnos_command.py:4]: Lenovo?

File: lib/ansible/modules/network/cnos/cnos_config.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: cnos_config
+version_added: "2.5"
```
Comment (gundalow) [inline lib/ansible/modules/network/cnos/cnos_config.py]: `version_added: "2.6"`?

File: lib/ansible/modules/network/cnos/cnos_command.py (hunk 1 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_command.py:4]: Working on this.

File: lib/ansible/modules/network/cnos/cnos_config.py (hunk 2 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_config.py]: Working on this.

Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_config.py]: Done

File: lib/ansible/modules/network/cnos/cnos_command.py (hunk 1 above)
Comment (amuraleedhar) [reply lib/ansible/modules/network/cnos/cnos_command.py:4]: Should I have to change any thing here ? The license statement has been reviewed by Legal team of Lenovo for enos_command.py, u remember ? They said as the original idea is that of Redhat, it has to come first and name of Lenovo should come next. Its same as I done for enos_command.py. And Lenovo is not Lenovo Inc. Its just Lenovo.

File: test/sanity/validate-modules/ignore.txt [hunk 3]
```diff
@@ -900,7 +900,8 @@ lib/ansible/modules/network/cnos/cnos_backup.py E322
 lib/ansible/modules/network/cnos/cnos_backup.py E323
 lib/ansible/modules/network/cnos/cnos_backup.py E326
 lib/ansible/modules/network/cnos/cnos_bgp.py E326
-lib/ansible/modules/network/cnos/cnos_command.py E326
+lib/ansible/modules/network/cnos/cnos_command.py E323
+lib/ansible/modules/network/cnos/cnos_config.py E323
```
Comment (mattclay) [inline test/sanity/validate-modules/ignore.txt]: New ignores should not be added. The underlying issue(s) should be fixed. If those issues are in existing module_utils or docs fragments they should be fixed in a separate PR first.

**Comment by mattclay** [comment]
CI failure in unit tests: https://app.shippable.com/github/ansible/ansible/runs/64759/3/tests

File: lib/ansible/modules/network/cnos/cnos_command.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/modules/network/cnos/cnos_command.py:4]: ah, ignore me. I read the diff incorrectly. 
What you have is correct
